### DISCUSSION TITLE
Bumping formula in custom tap that provides bottles from GitHub Releases

### DISCUSSION THREAD

**Discussion by rgoldberg** [Homebrew/discussions#5738] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

I am trying to maintain a custom tap that provides bottles from GitHub Releases.

What is the best way to use `brew bump-formula-pr` to easily update a formula from a custom tap?

I know of 3 places where the formula must be updated besides SHA 256 checksums.

`brew bump-formula-pr …` properly updates 2 of the 3, namely the `tag:` & `revision:` arguments for the `url` method, e.g.:

```bash
$ brew bump-formula-pr --dry-run --tag v1.8.8 --revision 4405807010987802c0967bbf349c08808062b824 mas-cli/tap/mas
==> replace /tag:(\s+")v1.8.7(?=")/ with "tag:\\1v1.8.8\\2"
==> replace "4405807010987802c0967bbf349c08808062b824" with "4405807010987802c0967bbf349c08808062b824"
```

The formula also uses a `root_url` in the `bottle` block to specify the URL to the online GitHub Releases directory containing the bottles:

```ruby
  bottle do
    root_url "https://github.com/mas-cli/mas/releases/download/v1.8.7"
    …
  end
```

Unfortunately, `brew bump-formula-pr …` doesn't update the version in that URL.

What is the best way to ensure that the bottle URLs are correct? Should I still use `root_url`?

When I tried to use `#{version}`:

```ruby
    root_url "https://github.com/mas-cli/mas/releases/download/v#{version}"
```

I got:

```bash
$ brew bump-formula-pr --write-only --no-audit --tag v1.8.8 --revision 4405807010987802c0967bbf349c08808062b824 mas-cli/tap/mas

Error: mas-cli/tap/mas: undefined local variable or method `version' for an instance of BottleSpecification
```

Will using a pull label (default being `pr-pull`) automatically update the SHA 256 checksums for the formula, or must I setup some other methodology to update SHA 256 checksums?

**Comment by SMillerDev** [comment, marked as answer]
> The formula also uses a root_url in the bottle block to specify the URL to the online GitHub Releases directory containing the bottles:

That will automatically update when you use the `pr-pull` label to pull and upload the new bottles. If it updated before the PR, it would not work since the release would not exist yet.

**Reply by rgoldberg** [reply]
Thanks for info. Will try it out.
